README.md

```diff
@@ -52,3 +52,43 @@ if err != nil {
     ...
 }
 ```
+
+### Verifying
+
+A `MessageVerifier` can be given a list of headers that are required in the message signature and a function that provides a public key and an optional expected hashing algorithm for a given `keyId`.
+
+```go
+func keyIDFetcher(keyID string) (crypto.PublicKey, crypto.Hash, error) {
+    keyBytes, err := os.ReadFile("path/to/public.key")
+    if err != nil {
+        ...
+    }
+
+    decoded, err := pem.Decode(keyBytes)
+    if err != nil {
+        ...
+    }
+
+    publicKey, err := x509.ParsePKIXPublicKey(decoded.Bytes)
+    if err != nil {
+        ...
+    }
+
+    rsaPK, ok := publicKey.(*rsa.PublicKey)
+    if !ok {
+        ...
+    }
+    return rsaPK, 0, nil
+}
+
+verifier := NewMessageVerifier(nil, keyIDFetcher)
+```
+
+Once you have a `MessageVerifier`, you can use it in a to verify an HTTP request. If the signature on the request contains the `digest` header, then the digest will be validated against the request `body`. If any `requiredHeaders` were provided, the signature headers will be validated against these. The `keyId` is then extracted and the signature is verified using the public key and optional hashing algorithm returned by the function provided to `MessageVerifier`.
```

Comment:
I think we missed a word in the first paragraph?

`...you can use it in a to verify an HTTP request`? 
Where can we us it? in a handler? middleware?

Reply:
Oh I think it's just supposed to say `... you can use it to verify an HTTP request`